kubernetes/provider.go

```diff
@@ -202,48 +202,52 @@ func providerConfigure(d *schema.ResourceData, terraformVersion string) (interfa
 	if err != nil {
 		return nil, err
 	}
+
+	hasConfig := false
 	if cfg == nil {
-		// Attempt to load in-cluster config
-		cfg, err = restclient.InClusterConfig()
-		if err != nil {
-			// Fallback to standard config if we are not running inside a cluster
-			if err == restclient.ErrNotInCluster {
-				cfg = &restclient.Config{}
-			} else {
-				return nil, fmt.Errorf("Failed to configure: %s", err)
-			}
-		}
+		cfg = &restclient.Config{}
+	} else {
+		hasConfig = true
 	}
 
 	// Overriding with static configuration
 	cfg.UserAgent = fmt.Sprintf("HashiCorp/1.0 Terraform/%s", terraformVersion)
 
 	if v, ok := d.GetOk("host"); ok {
+		hasConfig = true
 		cfg.Host = v.(string)
 	}
 	if v, ok := d.GetOk("username"); ok {
+		hasConfig = true
 		cfg.Username = v.(string)
 	}
 	if v, ok := d.GetOk("password"); ok {
+		hasConfig = true
 		cfg.Password = v.(string)
 	}
 	if v, ok := d.GetOk("insecure"); ok {
+		hasConfig = true
 		cfg.Insecure = v.(bool)
 	}
 	if v, ok := d.GetOk("cluster_ca_certificate"); ok {
+		hasConfig = true
 		cfg.CAData = bytes.NewBufferString(v.(string)).Bytes()
 	}
 	if v, ok := d.GetOk("client_certificate"); ok {
+		hasConfig = true
 		cfg.CertData = bytes.NewBufferString(v.(string)).Bytes()
 	}
 	if v, ok := d.GetOk("client_key"); ok {
+		hasConfig = true
 		cfg.KeyData = bytes.NewBufferString(v.(string)).Bytes()
 	}
 	if v, ok := d.GetOk("token"); ok {
+		hasConfig = true
 		cfg.BearerToken = v.(string)
```

Comment:
I still think this should do:
```
		cfg.BearerTokenFile = nil
```
no matter what to ensure no latent `BearerTokenFile` setting gets used.

It is possible the `cfg` created during:
```
		cfg, err = tryLoadingConfigFile(d)
```
has specified such a token file in the config file, and all that is wanted is to override it with a dynamically determined token.

Reply:
Good point, will do.

Reply:
Done.